server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -546,9 +546,11 @@ private static void parseObject(final ParseContext context, ObjectMapper mapper,
             ObjectMapper parentMapper = parentMapperTuple.v2();
             ObjectMapper.Dynamic dynamic = dynamicOrDefault(parentMapper, context);
             if (dynamic == ObjectMapper.Dynamic.STRICT) {
-                throw new StrictDynamicMappingException(mapper.fullPath(), currentFieldName);
-            } else if (dynamic == ObjectMapper.Dynamic.TRUE) {
-                Mapper.Builder builder = context.root().findTemplateBuilder(context, currentFieldName, XContentFieldType.OBJECT);
+                throw new StrictDynamicMappingException(dynamic.toString(), mapper.fullPath(), currentFieldName);
+            } else if (dynamic == ObjectMapper.Dynamic.TRUE || dynamic == ObjectMapper.Dynamic.STRICT_ALLOW_TEMPLATES) {
+                Mapper.Builder builder = context.root()
+                    .findTemplateBuilder(context, currentFieldName, XContentFieldType.OBJECT, dynamic, mapper.fullPath());
+
                 if (builder == null) {
```

Comment:
I would expect parsing to fail when there is no dynamic template matching the field, that is how I understand `strict` + `allow templates`

Reply:
This has been changed, right? Probably deserves the exception here?